Mu2eInterfaces/inc/ConditionsEntity.hh

```diff
@@ -15,7 +15,10 @@ namespace mu2e
   class ConditionsEntity
   {
   public:
+    ConditionsEntity() {}
     virtual ~ConditionsEntity();
+    ConditionsEntity(const ConditionsEntity&);
```

Comment:
Shouldn't the default c'tor and copy c'tor also be virtual?  I think they need to be override-able in the derived classes.

The same for Detector.hh and ProditionsEntity (and others?).

Reply:
Reminding myself about these dusty corners - ctor's can't be virtual because there is no vtable at construction.  I hear this as you are always constructing base or derived but never in a situation where you are constructing maybe dervied_1 or derived_2.  The assignment operator shouldn't generally be virtual since any derived class should provide a assignment operator with a signature of the derived type.  The user has to understand what types they are assigning. The derived default assignment will copy the base and derived class, and a custom derived assignment should deal with the base class members explicitly.  I see the virtual assignment operators of the base type do have a use case, but this is getting quite obscure.  It might be safest to delete the assignment operator of a base class unless some code is actually assigning between base class objects, which we shouldn't be doing in these cases.  Anyway, I will wait for your reply on the Detector.cc question below and fix both together.